e2e/tasks/fulfillOracleQuery.js

```diff
@@ -2,60 +2,77 @@
 
 const { ethers } = require('ethers');
 const { EvmPriceServiceConnection } = require('@pythnetwork/pyth-evm-js');
-const oracles = require('../deployments/oracles.json');
+const { parseError } = require('../parseError');
 
 const log = require('debug')(`e2e:${require('path').basename(__filename, '.js')}`);
 
-const PYTH_MAINNET_ENDPOINT = 'https://xc-mainnet.pyth.network';
-const PYTH_TESTNET_ENDPOINT = 'https://xc-testnet.pyth.network';
+const PYTH_MAINNET_ENDPOINT = 'https://hermes.pyth.network';
 
 const ERC7412_ABI = [
   'error OracleDataRequired(address oracleContract, bytes oracleQuery)',
   'error FeeRequired(uint feeAmount)',
   'function oracleId() view external returns (bytes32)',
   'function fulfillOracleQuery(bytes calldata signedOffchainData) payable external',
 ];
+const priceService = new EvmPriceServiceConnection(PYTH_MAINNET_ENDPOINT);
 
-async function fulfillOracleQuery({ wallet, symbol, isTestnet }) {
-  if (!(symbol in oracles)) {
-    throw new Error(`Symbol ${symbol} not found in oracles`);
+async function getPrice(feedId) {
+  const priceFeed = await priceService.getLatestPriceFeeds([feedId]);
+  if (priceFeed) {
+    return priceFeed[0].getPriceUnchecked();
   }
+  throw Error(`Price feed not found, feed id: ${feedId}`);
+}
 
-  const priceService = new EvmPriceServiceConnection(
-    isTestnet ? PYTH_TESTNET_ENDPOINT : PYTH_MAINNET_ENDPOINT
-  );
-
-  log({ symbol });
-
-  const oracle = oracles[symbol];
+function base64ToHex(str) {
+  const raw = Buffer.from(str, 'base64');
+  return '0x' + raw.toString('hex');
+}
 
-  const [offchainData] = await priceService.getPriceFeedsUpdateData([oracle.feedId]);
+async function fulfillOracleQuery({
+  wallet,
+  priceVerificationContractAddress,
+  oracleFeedId,
+  timestamp,
+}) {
+  log({ oracleFeedId });
+  const [offchainData] = await priceService.getVaa(oracleFeedId, timestamp);
+  const UPDATE_TYPE = 2;
   const offchainDataEncoded = ethers.utils.defaultAbiCoder.encode(
     ['uint8', 'uint64', 'bytes32[]', 'bytes[]'],
-    [1, oracle.staleness, [oracle.feedId], [offchainData]]
+    [UPDATE_TYPE, timestamp, [oracleFeedId], [base64ToHex(offchainData)]]
+  );
+
+  const PriceVerificationContract = new ethers.Contract(
+    priceVerificationContractAddress,
+    ERC7412_ABI,
+    wallet
   );
-  const OracleBTC = new ethers.Contract(oracle.address, ERC7412_ABI, wallet);
+  try {
+    const tx = await PriceVerificationContract.fulfillOracleQuery(offchainDataEncoded, {
+      value: ethers.BigNumber.from(1), // 1 wei,
+    });
+    await tx.wait();
+  } catch (error) {
+    parseError(error);
```

Comment:
log() missing?
Also instead of try/catch await bla().catch(parseError)

Reply:
I think it throws a nice enough message most of the time, can do log if you prefer that?
The reason was that something its the  .wait calls that fails.
I can change it do this if you prefer:
```ts
const tx =  await bla().catch(parseError)
await tx.wait().catch(parseError)
```

Reply:
Fixed

Reply:
Seen ti in some other place yeah not here